lib/geo_concerns/processors/gdal_projection.rb

```diff
@@ -0,0 +1,42 @@
+require 'open-uri'
+
+module GeoConcerns::Processors
+  class GdalProjection < Hydra::Derivatives::Processors::Processor
+    include Hydra::Derivatives::Processors::ShellBasedProcessor
+    attr_reader :source_path, :directives
+
+    def self.process(source_path, directives)
+      new(source_path, directives).reproject # XXX: clean up after ourselves
+    end
+
+    def initialize(source_path, directives)
+      @source_path = source_path
+      @directives = directives
+
+      @output_filename = output_file(File.extname(source_path).gsub(/^\./, ''))
+      @output_directory = File.dirname(@output_filename)
+
+      @srid ||= directives.fetch(:srid, 4326)
+
+      FileUtils.mkdir_p(@output_directory) unless File.directory?(@output_directory)
+    end
+
+    def reproject
+      gdalwarp
+    end
+
+    # reproject via GDAL, @see http://www.gdal.org/gdalwarp.html
+    def gdalwarp(resample = 'bilinear')
+      tempfn = File.join(@output_directory, File.basename(source_path))
+      # reproject with gdalwarp (must uncompress here to prevent bloat)
+      self.class.execute "gdalwarp -q -r #{resample} -t_srs EPSG:#{@srid} #{source_path} #{tempfn} -co 'COMPRESS=NONE'"
+      fail "gdalwarp failed to create #{tempfn}" unless File.size?(tempfn)
+
+      # compress tempfn with gdal_translate
+      self.class.execute "gdal_translate -q -a_srs EPSG:#{@srid} #{tempfn} #{@output_filename} -co 'COMPRESS=LZW'"
```

Comment:
Are there any use cases where a configuration file might be useful in specifying format-specific options regarding raster data sets?
I only ask given that the documentation for `gdal_translate` seems to indicate that there might be cases where formats such as JPEG2000 and MrSID require that there be an explicit expansion for the color banding:
> -expand gray|rgb|rgba:
>    (From GDAL 1.6.0) To expose a dataset with 1 band with a color table as a dataset with 3 (RGB) or 4 (RGBA) bands. Useful for output drivers such as JPEG, JPEG2000, MrSID, ECW that don't support color indexed datasets. The 'gray' value (from GDAL 1.7.0) enables to expand a dataset with a color table that only contains gray levels to a gray indexed dataset. 
(http://www.gdal.org/gdal_translate.html)

If this only affects cases where the processor would be writing _**to**_ JPEG2000, MrSID, etc., or, if these are edge cases, then please disregard this comment.